crates/biome_service/src/projects.rs

```diff
@@ -96,6 +96,15 @@ impl Projects {
             .map(|data| data.settings.clone())
     }
 
+    /// Unsafe version of [Projects::get_settings] that return an error
+    pub fn unwrap_settings(&self, project_key: ProjectKey) -> Result<Settings, WorkspaceError> {
```

Comment:
The naming and the comment of this function don't make a lot of sense to me. It's not "unsafe" (at least not in the Rust sense of the word), and it doesn't do any unwrapping either.

I'd rather we add `.ok_or_else(WorkspaceError::no_project)` to the regular `get_settings()` function, and any call site that wanted to have an `Option` instead could just call `.ok()` on the `Result` instead.